### PR TITLE
Add signature validation in download docs [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
Add the note about how to verify the signature for the release artifacts with the public key in OpenPGP

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

File: docs/download.md [hunk 1]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+* Import the public key: `gpg --import PUB_KEY`
+* Verift the signature: `gpg --verify rapids-4-spark_2.12-21.12.0.jar.asc rapids-4-spark_2.12-21.12.0.jar`
+
+Sinature verify output:
+
+	gpg: Good signature from "NVidia Spark (For the jars signature of spark rapids public release)<sw-spark@nvidia.com>"
+
+### Release Notes
+New functionality and performance improvements for this release include:
+
```
Comment (NvTimLiu) [inline docs/download.md:67]: Will add `functionality and performance improvements` here

File: docs/download.md [hunk 2]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+This package is built against CUDA 11.4 and has [CUDA forward
+compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html) enabled.  It is tested
+on V100, T4, A2, A10, A30 and A100 GPUs with CUDA 11.0-11.4.  For those using other types of GPUs which
+do not have CUDA forward compatibility (for example, GeForce), CUDA 11.2 is required. Users will
+need to ensure the minimum driver (450.80.02) and CUDA toolkit are installed on each Spark node.
+
+### Verify signature
+* Download the [RAPIDS Accelerator for Apache Spark 21.12.0 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar)
+  and the signature file [RAPIDS Accelerator for Apache Spark 21.12.0 jars.asc](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar.asc)
+* Search the public key via the user-Id [sw-spark@nvidia.com](sw-spark@nvidia.com) at keyserver [keys.openpgp.org](https://keys.openpgp.org) and download.
```
Comment (tgravescs) [inline docs/download.md:57]: perhaps just user-Id to email address to match text box on keys.openpgp.org.

or is there a direct link to it - like https://keys.openpgp.org/search?q=sw-spark%40nvidia.com we could put

File: docs/download.md [hunk 3]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+* Import the public key: `gpg --import PUB_KEY`
+* Verift the signature: `gpg --verify rapids-4-spark_2.12-21.12.0.jar.asc rapids-4-spark_2.12-21.12.0.jar`
```
Comment (tgravescs) [inline docs/download.md:59]: Verify misspelled

File: docs/download.md [hunk 4]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+
+Sinature verify output:
```
Comment (tgravescs) [inline docs/download.md:61]: signature misspelled

File: docs/download.md [hunk 5]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+
+	gpg: Good signature from "NVidia Spark (For the jars signature of spark rapids public release)<sw-spark@nvidia.com>"
```
Comment (tgravescs) [inline docs/download.md:63]: this is text it prints, any reason NVIDIA isn't all caps?

**Review by tgravescs — COMMENTED** [review]
only reviewed the verify signature section

File: docs/download.md [hunk 6]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+
+The plugin is tested on the following architectures:
+
+	GPU Architecture: NVIDIA V100, T4 and A2/A10/A30/A100 GPUs
+
+Software Requirements:
+
+	OS: Ubuntu 18.04, Ubuntu 20.04 or CentOS 7, CentOS 8
+
+	CUDA & NVIDIA Drivers*: 11.0-11.4 & v450.80.02+
```
Comment (GaryShen2008) [inline docs/download.md:32]: Should we add cuda 11.5 as well?

File: docs/download.md [hunk 7]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+
+*Some hardware may have a minimum driver version greater than v450.80.02+.  Check the GPU spec sheet
+for your hardware's minimum driver version.
+
+### Download v21.12.0
+* Download the [RAPIDS
+  Accelerator for Apache Spark 21.12.0 jar](https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.12.0/rapids-4-spark_2.12-21.12.0.jar)
+* Download the [RAPIDS cuDF 21.12.0 jar](https://repo1.maven.org/maven2/ai/rapids/cudf/21.12.0/cudf-21.12.0-cuda11.jar)
+
+This package is built against CUDA 11.4 and has [CUDA forward
```
Comment (GaryShen2008) [inline docs/download.md:48]: 21.12.0 will be built on CUDA 11.5, won't it?

File: docs/download.md [hunk 8]
```diff
@@ -18,6 +18,56 @@ cuDF jar, that is either preinstalled in the Spark classpath on all nodes or sub
...
+compatibility](https://docs.nvidia.com/deploy/cuda-compatibility/index.html) enabled.  It is tested
+on V100, T4, A2, A10, A30 and A100 GPUs with CUDA 11.0-11.4.  For those using other types of GPUs which
```
Comment (GaryShen2008) [inline docs/download.md:50]: Same for adding 11.5

File: docs/download.md (hunk 2 above)
Comment (NvTimLiu) [reply docs/download.md:57]: > 

Good point, let me update with the quick search link: https://keys.openpgp.org/search?q=sw-spark%40nvidia.com

Comment (NvTimLiu) [reply docs/download.md]: fixed,

File: docs/download.md (hunk 3 above)
Comment (NvTimLiu) [reply docs/download.md]: fixed,

File: docs/download.md (hunk 4 above)
Comment (NvTimLiu) [reply docs/download.md]: fixed,

File: docs/download.md (hunk 5 above)
Comment (NvTimLiu) [reply docs/download.md]: fixed,

File: docs/download.md (hunk 6 above)
Comment (NvTimLiu) [reply docs/download.md]: fixed,

File: docs/download.md (hunk 7 above)
Comment (NvTimLiu) [reply docs/download.md]: fixed,

File: docs/download.md (hunk 8 above)
Comment (NvTimLiu) [reply docs/download.md]: fixed,

**Comment by sameerz** [comment]
This should be superseded by PR #4307

**Comment by NvTimLiu** [comment]
CLOSE this PR since it was superseded by PR #4307

**Pull request closed without merging**